### DISCUSSION TITLE
Support deleting rows with LIMIT

### DISCUSSION THREAD

**Discussion by kevinyang372** [diesel-rs/diesel#2997] [Q&A]
I have a simple goal of selecting and deleting the first 10 rows from a database. The SQL statement for this would be
```DELETE FROM blocks WHERE blocks.id == x LIMIT 10;```

But when trying to convert this into diesel, I received a hard to read error saying the `Identifiable` trait bound is not satisfied:
```
diesel::delete(
     schema::blocks::dsl::blocks
              .filter(id.eq(x))
              .limit(10)
      )
      .execute(conn)?
```

**Comment by weiznich** [comment]
Can you please add more details to your question. `DELETE FROM blocks WHERE blocks.id == x LIMIT 10;` is no valid SQL for Postgresql for example.
